src/Workspaces/CSharp/Portable/Formatting/CSharpFormattingOptions.cs

```diff
@@ -4,145 +4,237 @@
 
         public static Option<bool> SpaceWithinExpressionParentheses { get; } = new Option<bool>(nameof(CSharpFormattingOptions), nameof(SpaceWithinExpressionParentheses), defaultValue: false,
-            storageLocations: new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpaceWithinExpressionParentheses"));
+            storageLocations: new OptionStorageLocation[] {
+                new EditorConfigStorageLocation("csharp_space_within_parentheses", (s) => DetermindIfSpaceOptionIsSet(s, SpacingWithinParenthesesOption.Expressions)),
+                new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpaceWithinExpressionParentheses")});
 
         public static Option<bool> SpaceWithinCastParentheses { get; } = new Option<bool>(nameof(CSharpFormattingOptions), nameof(SpaceWithinCastParentheses), defaultValue: false,
-            storageLocations: new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpaceWithinCastParentheses"));
+            storageLocations: new OptionStorageLocation[] {
+                new EditorConfigStorageLocation("csharp_space_within_parentheses", (s) => DetermindIfSpaceOptionIsSet(s, SpacingWithinParenthesesOption.TypeCasts)),
+                new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpaceWithinCastParentheses")});
 
         public static Option<bool> SpaceWithinOtherParentheses { get; } = new Option<bool>(nameof(CSharpFormattingOptions), nameof(SpaceWithinOtherParentheses), defaultValue: false,
-            storageLocations: new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpaceWithinOtherParentheses"));
+            storageLocations: new OptionStorageLocation[] {
+                new EditorConfigStorageLocation("csharp_space_within_parentheses", (s) => DetermindIfSpaceOptionIsSet(s, SpacingWithinParenthesesOption.ControlFlowStatements)),
+                new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpaceWithinOtherParentheses")});
 
         public static Option<bool> SpaceAfterCast { get; } = new Option<bool>(nameof(CSharpFormattingOptions), nameof(SpaceAfterCast), defaultValue: false,
-            storageLocations: new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpaceAfterCast"));
+            storageLocations: new OptionStorageLocation[] {
+                new EditorConfigStorageLocation("csharp_space_after_cast"),
+                new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpaceAfterCast")});
 
         public static Option<bool> SpacesIgnoreAroundVariableDeclaration { get; } = new Option<bool>(nameof(CSharpFormattingOptions), nameof(SpacesIgnoreAroundVariableDeclaration), defaultValue: false,
-            storageLocations: new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpacesIgnoreAroundVariableDeclaration"));
+            storageLocations: new OptionStorageLocation[] {
+                new EditorConfigStorageLocation("csharp_space_ignore_around_declaration_statements"),
```

Comment:
Is this one better done as csharp_space_around_declaration_statements = ignore or something? It feels like having the checkboxes that say "don't do this".